runner/query_run.go

```diff
@@ -0,0 +1,117 @@
+package runner
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/pganalyze/collector/input/postgres"
+	"github.com/pganalyze/collector/output/pganalyze_collector"
+	"github.com/pganalyze/collector/state"
+	"github.com/pganalyze/collector/util"
+)
+
+func SetupQueryRunnerForAllServers(ctx context.Context, servers []*state.Server, collectionOpts state.CollectionOpts, logger *util.Logger) {
+	if collectionOpts.ForceEmptyGrant {
+		return
+	}
+	for idx := range servers {
+		go func(server *state.Server) {
+			logger = logger.WithPrefixAndRememberErrors(server.Config.SectionName)
+			cleanupInterval := time.NewTicker(5 * time.Minute)
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case <-cleanupInterval.C:
+					cleanup(server)
+				default:
+					run(ctx, server, collectionOpts, logger)
+					time.Sleep(1 * time.Second)
+				}
+			}
+		}(servers[idx])
+	}
+}
+
+func run(ctx context.Context, server *state.Server, collectionOpts state.CollectionOpts, logger *util.Logger) {
+	for idx, query := range server.QueryRuns {
+		if !query.FinishedAt.IsZero() {
+			continue
+		}
+		server.QueryRunsMutex.Lock()
+		server.QueryRuns[idx].StartedAt = time.Now()
+		server.QueryRunsMutex.Unlock()
+		logger.PrintVerbose("Query run %d starting: %s", query.Id, query.QueryText)
+
+		db, err := postgres.EstablishConnection(ctx, server, logger, collectionOpts, query.DatabaseName)
+		if err != nil {
+			server.QueryRunsMutex.Lock()
+			server.QueryRuns[idx].FinishedAt = time.Now()
+			server.QueryRuns[idx].Error = err.Error()
+			server.QueryRunsMutex.Unlock()
+			continue
+		}
+		defer db.Close()
+
+		err = postgres.SetStatementTimeout(ctx, db, 60*1000)
+		if err != nil {
+			server.QueryRunsMutex.Lock()
+			server.QueryRuns[idx].FinishedAt = time.Now()
+			server.QueryRuns[idx].Error = err.Error()
+			server.QueryRunsMutex.Unlock()
+			continue
+		}
+
+		pid := 0
+		err = db.QueryRow("SELECT pg_backend_pid()").Scan(&pid)
+		if err == nil {
+			server.QueryRunsMutex.Lock()
+			server.QueryRuns[idx].BackendPid = pid
+			server.QueryRunsMutex.Unlock()
+		} else {
+			server.QueryRunsMutex.Lock()
+			server.QueryRuns[idx].FinishedAt = time.Now()
+			server.QueryRuns[idx].Error = err.Error()
+			server.QueryRunsMutex.Unlock()
+			continue
+		}
+
+		// The comment doesn't include /* pganalyze-collector */ so that query runs are reported separately in pganalyze
+		comment := fmt.Sprintf("/* pganalyze:no-alert,pganalyze-query-run:%d */ ", query.Id)
+		prefix := ""
+		result := ""
+		if query.Type == pganalyze_collector.QueryRunType_EXPLAIN {
+			prefix = "EXPLAIN (ANALYZE, BUFFERS, FORMAT JSON) "
```

Comment:
Do we want to add VERBOSE here? With auto_explain, we mention to enable verbose, and I think we do show things from verbose output if available.